x/poolmanager/router_test.go

```diff
@@ -2441,3 +2441,139 @@ func (suite *KeeperTestSuite) TestGetOsmoRoutedMultihopTotalSwapFee() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestCreateMultihopExpectedSwapOuts() {
+	tests := map[string]struct {
+		route                  []types.SwapAmountOutRoute
+		tokenOut               sdk.Coin
+		balancerPoolCoins      []sdk.Coins
+		concentratedPoolDenoms [][]string
+		poolCoins              []sdk.Coins
+		cumulativeRouteSwapFee sdk.Dec
+		sumOfSwapFees          sdk.Dec
+
+		expectedSwapOuts []sdk.Int
+		expectedError    bool
+	}{
+		"happy path: one route": {
+			route: []types.SwapAmountOutRoute{
+				{
+					PoolId:       1,
+					TokenInDenom: bar,
+				},
+			},
+			poolCoins: []sdk.Coins{sdk.NewCoins(sdk.NewCoin(foo, defaultInitPoolAmount), sdk.NewCoin(bar, defaultInitPoolAmount))},
+
+			tokenOut:         sdk.NewCoin(foo, defaultSwapAmount),
+			expectedSwapOuts: []sdk.Int{sdk.NewInt(1000002)},
+		},
+		"happy path: two route": {
+			route: []types.SwapAmountOutRoute{
+				{
+					PoolId:       1,
+					TokenInDenom: foo,
+				},
+				{
+					PoolId:       2,
+					TokenInDenom: bar,
+				},
+			},
+
+			poolCoins: []sdk.Coins{
+				sdk.NewCoins(sdk.NewCoin(foo, defaultInitPoolAmount), sdk.NewCoin(bar, defaultInitPoolAmount)), // pool 1.
+				sdk.NewCoins(sdk.NewCoin(bar, defaultInitPoolAmount), sdk.NewCoin(baz, defaultInitPoolAmount)), // pool 2.
+			},
+			tokenOut:         sdk.NewCoin(baz, sdk.NewInt(100000)),
+			expectedSwapOuts: []sdk.Int{sdk.NewInt(100002), sdk.NewInt(100001)},
```

Comment:
can we add one line comments to how we are getting numbers?

Reply:
Would also like explanations of where these numbers come from.

Reply:
added!